cfsdk/src/main/java/io/harness/cfsdk/CfClient.java

```diff
@@ -298,88 +298,102 @@ public boolean unregisterEventsListener(final EventsListener observer) {
 
     public boolean boolVariation(String evaluationId, boolean defaultValue) {
 
-        final Evaluation evaluation = getEvaluationById(
+        final Evaluation evaluation = getEvaluationById(evaluationId, target);
 
-                evaluationId,
-                target,
-                defaultValue
-        );
-
-        final Object value = evaluation.getValue();
-        if (value instanceof Boolean) {
-
-            return (Boolean) value;
+        if (evaluation == null) {
+            log.warn("Evaluation for {} is null, returning default value", evaluationId);
+            SdkCodes.warnDefaultVariationServed(evaluationId, target, String.valueOf(defaultValue));
+            return defaultValue;
         }
-        if (value instanceof String) {
 
-            return "true".equals(value);
+        final String value = evaluation.getValue();
+        if (value == null) {
+            log.warn("Evaluation was found for '{}', but the value was null, " +
+                    "returning default variation", evaluationId);
+            SdkCodes.warnDefaultVariationServed(evaluationId, target, String.valueOf(defaultValue));
         }
 
-        SdkCodes.warnDefaultVariationServed(evaluationId, target, String.valueOf(defaultValue));
-        return defaultValue;
+        pushToMetrics(evaluationId, evaluation);
```

Comment:
@erdirowlands is this still pushing a metric if we get a null ?

Reply:
Thanks. Great spot. I didn't add a return for this case

if (value == null)

Reply:
Fixed